File: in_Java/1379 Find a Corresponding Node of a Binary Tree in a Clone of That Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        Stack<TreeNode> st = new Stack<>();
        TreeNode cur = cloned;
        while(cur != null || !st.isEmpty()){
            while(cur != null){
                if(cur.val == target.val) return cur;
                st.push(cur);
                cur = cur.left;
            }
            cur = st.pop();
            cur = cur.right;
        }
        return cur;
    }
}

File: in_Python/1379 Find a Corresponding Node of a Binary Tree in a Clone of That Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def getTargetCopy(self, original, cloned, target):
        """
        :type original: TreeNode
        :type cloned: TreeNode
        :type target: TreeNode
        :rtype: TreeNode
        """
        q = collections.deque()
        q.append(cloned)
        while q:
            n = len(q)
            while n != 0:
                n -= 1
                h = q.popleft()
                if h.val == target.val:
                    return h
                if h.left:
                    q.append(h.left)
                if h.right:
                    q.append(h.right)
        return None